src/caikit_nlp_client/grpc_client.py

```diff
@@ -1,6 +1,9 @@
 import logging
 from dataclasses import dataclass
-from typing import Optional
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
```

Comment:
Why do we need this ?

Reply:
Type hints are not needed at runtime, so we can avoid importing those. `TYPE_CHECKING` is a special variable that is always `False` unless the code is being type checked with `mypy`